R/bambu-extendAnnotations-utilityExtend.R

```diff
@@ -598,43 +603,42 @@ includeOverlapReadClass <- function(candidateList, filteredOverlapList) {
                                         exonRangesCombinedFiltered,annotationGrangesList, prefix){
-    rowDataCombinedFiltered$newTxClass <- rowDataCombinedFiltered$readClassType
-    rowDataCombinedFiltered$newTxClass[rowDataCombinedFiltered$readClassType
+    rowDataCombinedFiltered$txClassDescription <- rowDataCombinedFiltered$readClassType
+    rowDataCombinedFiltered$txClassDescription[rowDataCombinedFiltered$readClassType
                                        == "unsplicedNew" & rowDataCombinedFiltered$novelGene] <-
       "newGene-unspliced"
-    rowDataCombinedFiltered$newTxClass[rowDataCombinedFiltered$readClassType
+    rowDataCombinedFiltered$txClassDescription[rowDataCombinedFiltered$readClassType
                                        == "allNew" & rowDataCombinedFiltered$novelGene] <-
       "newGene-spliced"
     extendedAnnotationRanges <- exonRangesCombinedFiltered
     mcols(extendedAnnotationRanges) <-
-      rowDataCombinedFiltered[, c("GENEID", "newTxClass","readCount", "NDR",
+      rowDataCombinedFiltered[, c("GENEID", "novelGene", "novelTranscript", "txClassDescription","readCount", "NDR",
                                   "relReadCount", "relSubsetCount")]
     equalRanges = rowDataCombinedFiltered[!is.na(rowDataCombinedFiltered$TXNAME),]
-  #remove extended ranges that are already present in annotation
-  extendedAnnotationRanges <- extendedAnnotationRanges[is.na(rowDataCombinedFiltered$TXNAME)]
-  annotationRangesToMerge <- annotationGrangesList
-  mcols(annotationRangesToMerge)$readCount <- 
-    rep(NA,length(annotationRangesToMerge))
-  mcols(annotationRangesToMerge)$newTxClass <- 
-    rep("annotation",length(annotationRangesToMerge))
-  mcols(annotationRangesToMerge)$NDR <- 
-    rep(NA,length(annotationRangesToMerge))
-  mcols(extendedAnnotationRanges) <- mcols(extendedAnnotationRanges)[,colnames(mcols(extendedAnnotationRanges))]
-  
-  #copy over stats to annotations from read classes
-  mcols(annotationRangesToMerge[equalRanges$TXNAME])$NDR = equalRanges$NDR
-  mcols(annotationRangesToMerge[equalRanges$TXNAME])$readCount = equalRanges$readCount
-  mcols(annotationRangesToMerge[equalRanges$TXNAME])$relReadCount = equalRanges$relReadCount
-  mcols(annotationRangesToMerge[equalRanges$TXNAME])$relSubsetCount = equalRanges$relSubsetCount
+    #remove extended ranges that are already present in annotation
+    extendedAnnotationRanges <- extendedAnnotationRanges[is.na(rowDataCombinedFiltered$TXNAME)]
+    annotationRangesToMerge <- annotationGrangesList
+    if(length(annotationGrangesList)){
+        mcols(annotationRangesToMerge)$readCount <- NA
+        mcols(annotationRangesToMerge)$txClassDescription <- "annotation"
+        mcols(annotationRangesToMerge)$novelTranscript <- FALSE
+        mcols(annotationRangesToMerge)$novelGene <- FALSE
+        mcols(annotationRangesToMerge)$NDR <- NA
+        mcols(extendedAnnotationRanges) <- mcols(extendedAnnotationRanges)[,colnames(mcols(extendedAnnotationRanges))]
```

Comment:
this line here I don't understand right now, seems like nothing is changed, but I might read this wrongly

Reply:
The line where you think nothing changed is correct, I just tidied it up because it looked ugly before.